### PR TITLE
cluster-init: add flag use-token-file-in-kubeconfig

### PR DESCRIPTION
**Pull request by hongkailiu**
This PR starts the work of https://issues.redhat.com/browse/DPTP-3152

The breaking change will be fixed by https://github.com/openshift/release/pull/32546

/cc @smg247

### CODE REVIEW

**Review by smg247 — CHANGES_REQUESTED** [review]

File: cmd/cluster-init/update_bootstrap_secrets.go [hunk 1]
```diff
@@ -208,6 +227,15 @@ func updateDPTPControllerManagerSecret(c *secretbootstrap.Config, o options) err
 }
 
 func updateRehearseSecret(c *secretbootstrap.Config, o options) error {
+	if o.useTokenFileInKubeconfig {
+		keyAndFieldToken := serviceAccountTokenFile(ciOperator, o.clusterName)
+		if err := updateSecretItemContext(c, "pj-rehearse", string(api.ClusterBuild01), keyAndFieldToken, secretbootstrap.ItemContext{
```
[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/cluster-init/update_bootstrap_secrets.go]: Done!

File: cmd/cluster-init/update_bootstrap_secrets.go [hunk 2]
```diff
@@ -191,6 +192,15 @@ func generateDockerConfigJsonSecretConfigTo(name string, namespace string, clust
 }
 
 func updatePodScalerSecret(c *secretbootstrap.Config, o options) error {
+	if o.useTokenFileInKubeconfig {
+		if err := updateSecretItemContext(c, podScaler, string(api.ClusterAPPCI),
+			fmt.Sprintf("sa.pod-scaler.%s.%s", o.clusterName, token), secretbootstrap.ItemContext{
```
[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/cluster-init/update_bootstrap_secrets.go]: Updated.

**Pull request merged**